spec.md

```diff
@@ -474,6 +474,122 @@ The following string-encoded integer literals all have the value `-42`:
 Radix-specified integer literals are only usable when constructing hardware
 integer literals.  Any use in place of an integer is disallowed.
 
+# Circuit Components
+
+Circuit components are the named objects which may be contained within a module.
+They are: nodes, wires, registers, ports, and submodule instances.
```

Comment:
what about "verification" statements (printfs, asserts, etc)

Reply:
This is a good question. Do you have an opinion on this?

Based on the direction I was leaning, I don't think they are. They:
- do not have a principal type, and they
- may not be connected